app/routes/bundesIdent/desktop.tsx

```diff
@@ -48,11 +49,22 @@ export const loader: LoaderFunction = async ({ request }) => {
     };
   }
 
-  return {};
+  const session = await getSession(request.headers.get("Cookie"));
+  const hasSurveyShown = session.get("hasSurveyShown") || false;
+  session.set("hasSurveyShown", hasSurveyShown);
```

Comment:
@pgurusinga Maybe you don't need to update the session (and sending the Set-Cookie header) here? A `const hasSurveyShown = Boolean(session.get("hasSurveyShown"))` might be enough. It's always false except after you set it to true.

Reply:
ah right, or could we just do `session.set("hasSurveyShown", false);` because this route is the entry point
1. set the `hasSurveyShown` as  `false`
2. commit as the header